### DISCUSSION TITLE
Segmentation fault while meshing with tet elements on a regular cubic domain

### DISCUSSION THREAD

**Discussion by SagarChandra** [neperfepx/neper#516] [Q&A] [answered]
Hello All,

I am using Neper 4.0.2.

I want to have a polycrystal FE mesh with fine central grain and coarser outer grains. The morphology of the polycrystal RVE is cubical domain with regular shaped cubic grains. I did:

```
$ neper -T -n from_morpho -id 1 -domain "cube(6,6,6)" -morpho "cube(6,6,6)"
$ neper -M nfrom_morpho-id1.tess  -rcl "(body>0)?0.5:1.2" -format inp
```
The output terminates in Segmentation fault:

```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.0.2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt
Info   : Running on 2 threads.
Info   : <http://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/internet-user/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] nfrom_morpho-id1.tess
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `nfrom_morpho-id1.tess'...
Info   :     [i] Parsed file `nfrom_morpho-id1.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.5) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...   0% (0|0/ 0%| 0%| 0%)Segmentation fault
```

Please note that this error does not occur while meshing the same domain with hex elements.

Any help will be appreciated.

Thanks and Regards,
Sagar

**Comment by smallrain1996** [comment]
It is strongly recommended to update to the latest version, currently 4.4.2-29

**Reply by smallrain1996** [reply]
```
git clone https://github.com/rquey/neper.git
cd neper/src
mkdir build
cd build
cmake ..
make
sudo make install
make test
```

**Reply by SagarChandra** [reply]
Thank you so much for your reply. Upgrading to the latest version solved the problem.

**Comment by rquey** [comment, marked as answer]
Did you install Gmsh? Did you run the tests?

**Reply by SagarChandra** [reply]
Yes Professor Quey, i did install Gmsh (ver 2.10.1); however i did not run the make test command before executing neper. When upgarded to the new version of neper 4.4.2-29, i did not run make test either after installation; however the error of segmentation fault disappeared and the output mesh was as desired.

Is it really necessary to run make test after each upgrade?

**Reply by rquey** [reply]
It is recommended to run the tests to make sure that everything is in good order, but if you don't and it works, then it's probably fine. Tests help to diagnose installation problems.

PS. Version 2.10.1 for Gmsh is very old.

**Reply by SagarChandra** [reply]
Ok. Thank you for your suggestions.

**Reply by Geo572** [reply]
Hello @SagarChandra to upgrade Neper to the latest version you just use the following commands or some others.
cd neper/src
mkdir build
cd build
cmake ..
make
sudo make install
make test


Thank you
